src/tasks/gitea/gitea-drone-oauth.ts

```diff
@@ -0,0 +1,224 @@
+  } else if (oauthApps.length == 0) {
+    console.log("Gitea doesn't have any oauth application defined")
+  } else if (!oauthApps.some((e) => e.clientId === Buffer.from(remoteSecret.clientId, 'base64').toString())) {
+    console.log('OAuth data did not match with expected secret')
+  } else {
+    return true
+  }
+  return false
+}
+async function createK8SSecret(apiClient: k8s.CoreV1Api, result: DroneSecret) {
+  const secret: k8s.V1Secret = new k8s.V1Secret()
+  secret.metadata = new k8s.V1ObjectMeta()
+  secret.metadata.name = k8sSecretName
+  secret.metadata.namespace = k8sNamespace
+  secret.data = {
+    clientId: Buffer.from(result.clientId).toString('base64'),
+    clientSecret: Buffer.from(result.clientSecret).toString('base64'),
+  }
+
+  await apiClient.createNamespacedSecret(k8sNamespace, secret)
+  console.log(`New secret ${k8sSecretName} has been created in the namespace ${k8sNamespace}`)
+}
+async function main() {
+  const giteaUrl: string = env.GITEA_URL.endsWith('/') ? env.GITEA_URL.slice(0, 1) : env.GITEA_URL
+  const droneLoginUrl: string = giteaUrl.replace('gitea', 'drone') + '/login'
+
+  const kc = new k8s.KubeConfig()
+  kc.loadFromDefault()
+  const apiClient = kc.makeApiClient(k8s.CoreV1Api)
+
+  const user = new UserApi(env.GITEA_USER, env.GITEA_PASSWORD, `${giteaUrl}/api/v1`)
+  const oauthOpts = new CreateOAuth2ApplicationOptions()
+  oauthOpts.name = 'otomi-drone'
+  oauthOpts.redirectUris = [droneLoginUrl]
+  const remoteSecret: DroneSecret = await getSecret(apiClient)
+  const oauthAppsResponse = await user.userGetOauth2Application()
+  const oauthApps: OAuth2Application[] = oauthAppsResponse.body.filter((x) => x.name === oauthOpts.name)
+
+  // If secret exists (correctly) and oauth apps are (still) defined, everything is good
+  if (isSecretValid(remoteSecret, oauthApps)) {
+    console.log('Gitea Drone OAuth secret exists')
+    const oauthData: DroneSecret = {
+      clientId: Buffer.from(remoteSecret.clientId, 'base64').toString(),
+      clientSecret: Buffer.from(remoteSecret.clientSecret, 'base64').toString(),
+    }
+    await authorizeOAuthApp(giteaUrl, droneLoginUrl, oauthData)
+    return
+  }
+
+  // Otherwise, clear (if necessary)
+  try {
+    await apiClient.deleteNamespacedSecret(k8sSecretName, k8sNamespace)
+  } catch (e) {
+    // Secret didn't exist in
+  }
+  if (oauthApps.length > 0) {
+    for (const oauthApp of oauthApps) {
+      user.userDeleteOAuth2Application(oauthApp.id)
+    }
+  }
+
+  const createOauthAppsResponse = await user.userCreateOAuth2Application(oauthOpts)
+  const result: OAuth2Application = createOauthAppsResponse.body
+  console.log('OAuth app has been created')
+  const oauthData: DroneSecret = {
+    clientId: result.clientId,
+    clientSecret: result.clientSecret,
+  }
+
+  await authorizeOAuthApp(giteaUrl, droneLoginUrl, oauthData)
+  console.log('OAuth app has been authorized')
+
+  createK8SSecret(apiClient, oauthData)
```

Comment:
please create the client above (lazily) with:
```ts
let apiClient = undefined

getApiClient(): k8s.CoreV1Api {
  if (apiClient) return apiClient
  const kc = new k8s.KubeConfig()
  kc.loadFromDefault()
  apiClient = kc.makeApiClient(k8s.CoreV1Api)
  return apiClient
}
```

and then:
```suggestion
  createK8SSecret(getApiClient(), oauthData)
```